File: AMF5/src/test/java/javaPlatform/ShapeSerializationTest.java
package javaPlatform;

import amf.apicontract.client.platform.OASConfiguration;
import amf.apicontract.client.platform.AMFElementClient;
import amf.core.client.platform.model.DataTypes;
import amf.core.client.platform.model.domain.Shape;
import amf.shapes.client.platform.model.domain.AnyShape;
import amf.shapes.client.platform.model.domain.ScalarShape;
import org.junit.Test;

import java.io.IOException;

import static javaPlatform.StringEquals.*;

public class ShapeSerializationTest implements FileReader {

    @Test
    public void serializeToJsonSchema() throws IOException {
        AnyShape shape = (AnyShape) new ScalarShape()
                .withMinimum(2)
                .withMaximum(7)
                .withDataType(DataTypes.Double())
                .withId("something");
        AMFElementClient client = OASConfiguration.OAS20().elementClient();
        String jsonSchemaShape = client.toJsonSchema(shape);
        String expected = readResource("/expected/draft4-scalar-schema.json");
        assertIgnoringWhitespace(jsonSchemaShape, expected);
    }
}

File: AMF5/src/test/scala/scalaPlatform/ShapeSerializationTest.scala
package scalaPlatform

import amf.apicontract.client.scala.OASConfiguration
import amf.core.client.platform.model.DataTypes
import amf.shapes.client.scala.model.domain.ScalarShape
import org.junit.runner.RunWith
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should.Matchers
import org.scalatestplus.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class ShapeSerializationTest extends AsyncFlatSpec with Matchers with FileReader {

  "AMFElementClient" should "serialize to json schema" in {
    val shape = ScalarShape()
      .withId("someId")
      .withDataType(DataTypes.Double)
      .withMinimum(2)
      .withMaximum(7)
    val client = OASConfiguration.OAS30().elementClient()
    val jsonSchemaString = client.toJsonSchema(shape)
    val expected = readResource("/expected/draft4-scalar-schema.json")
    jsonSchemaString shouldEqual expected
  }
}
